extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -0,0 +1,295 @@
+
+        messages = self._construct_chat_history(prompt, aiconfig, params)
+        messages.append({"role": "user", "parts": [{"text": resolved_prompt}]})
+
+        completion_data["contents"] = messages
+
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_complete", __name__, {"output": completion_data}
+            )
+        )
+        return completion_data
+
+    async def run_inference(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        options: InferenceOptions,
+        parameters,
+    ) -> list[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            ExecuteResult: The response from the model.
+        """
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_run_start",
+                __name__,
+                {"prompt": prompt, "options": options, "parameters": parameters},
+            )
+        )
+
+        # Auth check. don't need to explicitly set the key as long as this is set as an env var. genai.configure() will pick it up
+        get_api_key_from_environment("GOOGLE_API_KEY")
+        genai.configure()
+
+        # TODO: check and handle api key here
+        completion_data = await self.deserialize(prompt, aiconfig, parameters)
+
+        stream = True  # Default value
+        if options is not None and options.stream is not None:
+            stream = options.stream
+        elif "stream" in completion_data:
+            stream = completion_data["stream"]
+        completion_data["stream"] = stream
+
+        response = await self.model.generate_content_async(**completion_data)
+
+        outputs = None
+        if stream:
+            outputs = await construct_stream_outputs(response, options)
+        else:
+            outputs = construct_regular_outputs(response)
+
+        prompt.outputs = outputs
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent("on_run_complete", __name__, {"result": prompt.outputs})
+        )
+        return prompt.outputs
+
+    def get_output_text(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        output: Optional[Output] = None,
+    ) -> str:
+        if not output:
+            output = aiconfig.get_latest_output(prompt)
+
+        if not output:
+            return ""
+
+        if output.output_type == "execute_result":
+            output_data = output.data
+            if isinstance(output_data, str):
+                return output_data
+            else:
+                raise Exception("Not Implemented")
```

Comment:
Shouldn't happen, right?

Reply:
Not expected, but I hadn't thoroughly understood the expected result types. I put this in as a just in case

Reply:
Seems like a `ValueError`

Reply:
Agreed let's update the exception at least